### DISCUSSION TITLE
DepthAI support for Gen2 API?

### DISCUSSION THREAD

**Discussion by gtech888AU** [emgucv/emgucv#596] [General]
I notice the latest release of emgucv has support for DepthAI v1.00 which was released in Feb 2021.  

These OAK cameras have had a very successful Kickstarter campaign and are currently undergoing another Kickstarter campaign for lower entry price depth camera.  (https://www.kickstarter.com/projects/opencv/opencv-ai-kit-oak-depth-camera-4k-cv-edge-object-detection)

Since this release of DepthAI v1.00, the API has changed to version 2, which has vastly increased the flexibility by which users can connect the various nodes together in the pipeline, and has also introduced on-device edge detection, scripting, object tracking etc .  From a usability perspective this new API is a game-changer, and really has made the v1 API obsolete.  The new API is currently at v2.10.  

I was very excited to see support for DepthAI within emgucv.  After drilling into it I noticed it only currently supported the older version.  I am wondering if there is currently some emgucv beta code which wraps this new API?  

If not, does emgu plan to support the new API in the future?

**Comment by Miticoval** [comment]
Hi, I'm upping this old post to ask news about this topic.

**Comment by emgucv** [comment]
The current master branch, as well as the up-coming Emgu CV 4.7 release will be based on DepthAI v2.15.4. There are a lot of changes in the DEPTH AI api and the current API 2.x coverage is not as good as teh API v1.x coverage in v4.6. A lot more development is needed to make the API 2.x coverage sufficient.
